src/zcl_aff_test_types.clas.abap

```diff
@@ -67,6 +67,15 @@ CLASS zcl_aff_test_types DEFINITION
         default_link TYPE default_link,
       END OF struc_link_wrong_type.
 
+    TYPES:
+     "! <p class="shorttext">Structure With Pattern Annotation</p>
+     "! Structure with pattern annotation
+       BEGIN OF string_pattern,
+         "! <p class="shorttext">String with pattern</p>
+         "! description
+         "! $pattern '[a-Z]*'
+         string_pattern TYPE string,
+       END OF string_pattern.
```

Comment:
Could you add an example like
```suggestion
    TYPES:
    "! $pattern '[a-Z]*'
    ty_string TYPE string.
    
    TYPES:
     "! <p class="shorttext">Structure With Pattern Annotation</p>
     "! Structure with pattern annotation
       BEGIN OF string_pattern,
         "! <p class="shorttext">String with pattern</p>
         "! description
         string_pattern TYPE ty_string,
       END OF string_pattern.
```
and a corresponding unit test? This test should expect the same as yours did but this one might fail